src/uu/ls/src/ls.rs

```diff
@@ -1315,20 +1330,37 @@ impl PathData {
             Dereference::None => false,
         };
 
+        let de = match dir_entry {
+            Some(de) => OnceCell::from(de.ok()),
+            None => OnceCell::new(),
+        };
+
         let ft = match file_type {
             Some(ft) => OnceCell::from(ft.ok()),
             None => OnceCell::new(),
         };
 
+        let md = match metadata {
+            Some(md) => {
+                if !must_dereference {
+                    OnceCell::from(md.ok())
+                } else {
+                    OnceCell::new()
+                }
+            }
+            None => OnceCell::new(),
+        };
```

Comment:
```suggestion
        let md = match metadata {
            Some(md) if !must_dereference => OnceCell::from(md.ok()),
            _ => OnceCell::new(),
        };
```
Also here, it'd be nice to have a small explanation about `must_dereference`

Reply:
We can just eliminate this now.